docker-compose.yaml

```diff
@@ -19,79 +19,126 @@ x-default: &node
+  AWS_S3_BUCKET:
+  AWS_S3_PREFIX:
+  GH_APP_ID:
+
+  IPTS_PRD_SERVICE_NAME:
+  IPTS_PRD_HOSTNAME:
+  IPTS_PRD_USERNAME:
+  IPTS_PRD_PORT:
+  IPTS_TST_SERVICE_NAME:
+  IPTS_TST_HOSTNAME:
+  IPTS_TST_USERNAME:
+  IPTS_TST_PORT:
 
 services:
-  spark-node-master:
+  spark-node-master-prod:
     <<: *node
-    container_name: spark-node-master
+    profiles: ["prod", ""] # empty string here runs when no profile is specified
+    image: ${CCAO_REGISTRY_URL}/service-spark-iasworld:latest
+    container_name: spark-node-master-prod
     hostname: spark-node-master
+    restart: unless-stopped
     environment:
-      - SPARK_MODE=master
-
-      - AWS_SHARED_CREDENTIALS_FILE=/run/secrets/AWS_CREDENTIALS
-      - AWS_S3_BUCKET
-      - AWS_S3_PREFIX
-      - SPARK_LOCAL_STORAGE_ENCRYPTION_ENABLED=no
-      - SPARK_PUBLIC_DNS=${SPARK_PUBLIC_IP}
-      - SPARK_RPC_AUTHENTICATION_ENABLED=no
-      - SPARK_RPC_ENCRYPTION_ENABLED=no
-      - SPARK_SSL_ENABLED=no
-      - SPARK_USER=spark
-      - IPTS_PRD_SERVICE_NAME
-      - IPTS_PRD_HOSTNAME
-      - IPTS_PRD_USERNAME
-      - IPTS_PRD_PORT
-      - IPTS_TST_SERVICE_NAME
-      - IPTS_TST_HOSTNAME
-      - IPTS_TST_USERNAME
-      - IPTS_TST_PORT
-      - GH_APP_ID
+      <<: [*spark-env]
+      SPARK_MODE: master
+      SPARK_UI_PORT: 4040
+      SPARK_EXECUTOR_MEMORY: 96g
+      SPARK_MASTER_WEBUI_PORT: 8080
     ports:
       - 4040:4040
       - 8080:8080
+    networks:
+      - sparknet-prod
 
-  spark-node-worker:
+  spark-node-worker-prod:
     <<: *node
-    container_name: spark-node-worker
+    profiles: ["prod", ""]
+    image: ${CCAO_REGISTRY_URL}/service-spark-iasworld:latest
+    container_name: spark-node-worker-prod
     hostname: spark-node-worker
+    restart: unless-stopped
+    depends_on:
+      - spark-node-master-prod
     environment:
-      - SPARK_MODE=worker
-      - SPARK_MASTER_URL=spark://spark-node-master:7077
-      - SPARK_WORKER_MEMORY=96G
-      - SPARK_WORKER_CORES=28
-      - SPARK_WORKER_WEBUI_PORT=9090
-
-      - AWS_SHARED_CREDENTIALS_FILE=/run/secrets/AWS_CREDENTIALS
-      - AWS_S3_BUCKET
-      - AWS_S3_PREFIX
-      - SPARK_PUBLIC_DNS=${SPARK_PUBLIC_IP}
-      - SPARK_RPC_AUTHENTICATION_ENABLED=no
-      - SPARK_RPC_ENCRYPTION_ENABLED=no
-      - SPARK_LOCAL_STORAGE_ENCRYPTION_ENABLED=no
-      - SPARK_SSL_ENABLED=no
-      - SPARK_USER=spark
-      - IPTS_PRD_SERVICE_NAME
-      - IPTS_PRD_HOSTNAME
-      - IPTS_PRD_USERNAME
-      - IPTS_PRD_PORT
-      - IPTS_TST_SERVICE_NAME
-      - IPTS_TST_HOSTNAME
-      - IPTS_TST_USERNAME
-      - IPTS_TST_PORT
-      - GH_APP_ID
+      <<: [*spark-env]
+      SPARK_MODE: worker
+      SPARK_WORKER_MEMORY: 96G
+      SPARK_WORKER_CORES: 32
+      SPARK_WORKER_WEBUI_PORT: 9090
     ports:
       - 9090:9090
+    networks:
+      - sparknet-prod
+
+  spark-node-master-dev:
+    <<: *node
+    profiles: ["dev"]
+    image: ${CCAO_REGISTRY_URL}/service-spark-iasworld:dev
```

Comment:
Note the different image tag for the dev containers. This will force a `docker compose build` on first run, which ensures the `dev` container reflects the new user and any changes made to the Dockerfile or dependencies.